### PR TITLE
Feature/sw 79 finish testcase for lenscalibration

### PR DESCRIPTION
**Pull request by JFSolution**
added testcases for boarddetection, lenscalibration and marker detection
Jira Ticket https://mr-beam.atlassian.net/browse/SW-79

### CODE REVIEW

File: octoprint_camera/camera.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def get_latest_img(self):
             self._logger.warning("Camera not initalised, cannot return an image")
             return None
 
+    #
+
```
Comment (amrsoll) [inline octoprint_camera/camera.py]: Please rm

File: octoprint_camera/static/js/calibration/corner_calibration.js [hunk 2]
```diff
@@ -233,8 +233,10 @@ $(function () {
             if (!self.cornerCalibrationActive()) {
                 self.camera.loadAvaiableCorrection();
                 if (self.camera.availablePicTypes.corners()) {
+                    console.log('get image corner - corner');
                     self.camera.getImage(GET_IMG.last, GET_IMG.pic_corner);
                 } else {
+                    console.log('get image corner - plain')
```
Comment (amrsoll) [inline octoprint_camera/static/js/calibration/corner_calibration.js]: Please rm logs if they are no longer needed

File: octoprint_camera/static/js/calibration/watterott/calibration_qa.js [hunk 3]
```diff
@@ -25,9 +27,40 @@ $(function () {
                 self.qa_image_loaded(false);
             }
         });
-        self.croppedImgWidth = 500;
+        self.croppedImgWidth = "500";
```
Comment (amrsoll) [inline octoprint_camera/static/js/calibration/watterott/calibration_qa.js]: Why is it converted into a string?

File: octoprint_camera/static/js/camera.js [hunk 4]
```diff
@@ -419,6 +419,8 @@ $(function () {
                 .join(" ");
         });
         self.zObjectImgTransform = ko.computed(function () {
+            console.log('zobjectimg trans', self.imgHeightScale(), self.imgTranslate(), self.workingAreaWidthMM(), self.workingAreaHeightMM(), 'imgheightscale params(',self.cornerMargin() *
+                (1 - self.objectZ() / self.maxObjectHeight), ';', self.cornerMargin(), self.objectZ(), self.maxObjectHeight, ')', self.defaultMargin);
```
Comment (amrsoll) [inline octoprint_camera/static/js/camera.js]: Same as before when it comes to logging - please rm

File: octoprint_camera/static/js/camera.js [hunk 5]
```diff
@@ -136,7 +136,7 @@ $(function () {
                     self.availablePicTypes[m](data.available_corrections.includes(m));
                 });
             };
-
+            // self.availablePicTypes['corners'](true);
```
Comment (amrsoll) [inline octoprint_camera/static/js/camera.js]: Please rm

File: octoprint_camera/static/js/calibration/watterott/calibration_qa.js [hunk 6]
```diff
@@ -25,9 +27,40 @@ $(function () {
...
+            console.log('zobjectimg trans', self.camera.imgHeightScale(), self.camera.imgTranslate(), self.workingAreaWidthMM(), self.workingAreaHeightMM(), 'imgheightscale params(',self.cornerMargin() *
+                (1 - self.objectZ() / self.maxObjectHeight()), ';', self.cornerMargin(), self.objectZ(), self.maxObjectHeight(), ')', self.defaultMargin());
+            return (
+                "scale(" +
+                (1 + 2 * self.imgHeightScale()) +
+                ") translate(" +
+                self.imgTranslate() +
+                ")"
+            );
+        });
```
Comment (amrsoll) [inline octoprint_camera/static/js/calibration/watterott/calibration_qa.js]: Please do not introduce duplication. If you need these functions and observables, then import them from other viewmodels. In this case, use the `cameraViewmodel` - `self.camera`.

Comment (amrsoll) [reply octoprint_camera/static/js/calibration/watterott/calibration_qa.js]: You can also create a new pointer if you want the qa screen viewmodel to have `self.some_attr` :

```js
self.some_attr = self.camera.some_attr
```

This way you keep all the `ko` bindings that are used elsewhere.

Example:

```js
self.zObjectImgTransform = self.camera.zObjectImgTransform
```

File: tests/README.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+##prepare the test
+clone repository `git@github.com:mrbeam/test_rsc.git`, this contains the files needed for the test
```
Comment (amrsoll) [inline tests/README.md:2]: please let us know if we need to clone it somewhere special or make a symlink from this repo

**Review by amrsoll — CHANGES_REQUESTED** [review]
Great :) Please be careful about duplicity though.

File: octoprint_camera/static/js/calibration/watterott/calibration_qa.js (hunk 3 above)
Comment (JFSolution) [reply octoprint_camera/static/js/calibration/watterott/calibration_qa.js]: hmm I don't know, this was not intended, will change it back

File: octoprint_camera/camera.py (hunk 1 above)
Comment (JFSolution) [reply octoprint_camera/camera.py]: removed

File: octoprint_camera/static/js/camera.js (hunk 4 above)
Comment (JFSolution) [reply octoprint_camera/static/js/camera.js]: done

**Review by amrsoll — APPROVED** [review]
Great!

**Pull request merged**